### DISCUSSION TITLE
How fluent-bit works when output is unavailable?

### DISCUSSION THREAD

**Discussion by DonVietnam** [fluent/fluent-bit#7883] [General]
For example, I need to implement the following pipeline, data with logs is sent to fluentbit, then it sends data to Kafka topic, what happens if the Kafka cluster falls and is temporarily unavailable? Will fluent bit save the data and as soon as Kafka becomes available again, will it start sending them further? Or will the data be lost? If there is a place to read about this situation, I will be grateful for the link.

**Comment by patrick-stephens** [comment]
Did you check the docs?
- https://docs.fluentbit.io/manual/administration/backpressure
- https://docs.fluentbit.io/manual/administration/buffering-and-storage
